packages/repo/src/parse.ts

```diff
@@ -1,21 +1,34 @@
 import { check, cborDecode } from '@atproto/common'
+import { cborToLexRecord, RepoRecord } from '@atproto/lexicon'
 import { CID } from 'multiformats/cid'
 import BlockMap from './block-map'
 import { MissingBlockError, UnexpectedObjectError } from './error'
 
-export const getAndParse = async <T>(
+export const getAndParseRecord = async (
+  blocks: BlockMap,
+  cid: CID,
+): Promise<{ record: RepoRecord; bytes: Uint8Array }> => {
+  const bytes = blocks.get(cid)
+  if (!bytes) {
+    throw new MissingBlockError(cid, 'record')
+  }
+  const record = await cborToLexRecord(bytes)
+  return { record, bytes }
+}
```

Comment:
What are your thoughts on keeping lex records pretty much entirely out of the repo/storage code, and working exclusively with ipld values there.  In a more lexicon-aware such as an xrpc server (i.e. the pds) you would end-up calling `ipldToLex()` and `lexToIpld()` when talking to the repo or storage.  I know you spent a lot of time considering this kind of thing, so mostly interested in your current thinking on it.

Reply:
Yeah I'm not opposed to going back to that. I just found that it was just prone to a lot of footguns. A lot of places where you have to make sure you correctly serialize your data before sending it to the repo. & the type system has a hard time calling all of those out.

I'm also not sure if it's bad for the repo to be _lightly_ aware of lex semantics. Like maybe you aren't allowed to put _anything_ at repo leaves. Maybe they _have_ to be a lexicon-record-deserealizable value 🤔

Reply:
That makes sense, sounds good.  Would definitely want good support for that in the type system, so if that's not coming very naturally then that's good enough reason to punt on it for me.  Plus we can always come back to it, I don't see it as a super high stakes thing 👍